auslib/db.py

```diff
@@ -403,14 +403,12 @@ def _prepareInsert(self, trans, changed_by, **columns):
         if self.versioned:
             data['data_version'] = 1
         query = self._insertStatement(**data)
-
-        if self.onInsert:
-            self.onInsert(self, "INSERT", changed_by, query, trans)
-
         ret = trans.execute(query)
         if self.history:
             for q in self.history.forInsert(ret.inserted_primary_key, data, changed_by):
                 trans.execute(q)
+        if self.onInsert:
+            self.onInsert(self, "INSERT", changed_by, query, trans, *ret.inserted_primary_key)
```

Comment:
How primary keys can be composable, may be a good idea pass the entire list as parameter and use it on the target callback, instead of dereference it.